### PR TITLE
Change default RSA key size to 2048 bit

### PR DESCRIPTION
**Pull request by kroeckx**
Fixes: #8737

### CODE REVIEW

**Comment by kroeckx** [comment]
I've forced pushed a new commit

File: crypto/dsa/dsa_pmeth.c [hunk 1]
```diff
@@ -20,7 +20,7 @@
 
 typedef struct {
     /* Parameter gen parameters */
-    int nbits;                  /* size of p in bits (default: 1024) */
+    int nbits;                  /* size of p in bits (default: 2048) */
     int qbits;                  /* size of q in bits (default: 160) */
```
Comment (bernd-edlinger) [inline crypto/dsa/dsa_pmeth.c]: this comment needs adjustment too

**Comment by levitte** [comment]
BTW, is this a security-related fix per se? I'm thinking of the 1.1.0 and 1.0.2 labels...

**Comment by kroeckx** [comment]
You really shouldn't be using 1024 bit keys anymore. People using
genpkey still get 1024 bit keys, while things like genrsa,
dhparam, and req where already changed to 2048 in 1.0.2.

**Comment by levitte** [comment]
> ... while things like genrsa, dhparam, and req where already changed to 2048 in 1.0.2.

Thanks, that's justification enough

**Comment by mattcaswell** [comment]
@kroeckx can this be merged now?

**Comment by kroeckx** [comment]
I'll do it if I have some time, but feel free to do it.

**Comment by slontis** [comment]
Needs a rebase..

**Pull request closed without merging**